### PR TITLE
Add devcontainer

### PR DESCRIPTION
**Pull request by rinx**
Signed-off-by: Rintaro Okamura <rintaro.okamura@gmail.com>

### Description:

To enable GitHub Codespaces, I added .devcontainer/devcontainer.json & dockers/dev/Dockerfile.

### Related Issue:

Nothing.

### How Has This Been Tested?:

Nothing.

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [X] Update CI [type/ci] ?

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "terminal.integrated.shell.linux": "/bin/bash",
+    "go.useGoProxyToCheckForToolUpdates": false,
+    "go.autocompleteUnimportedPackages": true,
+    "go.buildOnSave": true,
+    "go.vetOnSave": true,
+    "go.lintOnSave": true,
+    "go.testOnSave": true,
+    "go.gopath": "/go"
+  },
+  "extensions": [
```
Comment (rinx) [inline .devcontainer/devcontainer.json:18]: @kmrmt Do you have any recommends of extensions for Go development in VSCode?

Comment (kmrmt) [reply .devcontainer/devcontainer.json:18]: LGTM :+1:

**Comment by rinx** [comment]
@kmrmt @kpango @vankichi Now it's ready to use. Please try it on your Codespaces and approve if it's okay.

**Review by kpango — DISMISSED** [review]
LGTM

**Comment by kpango** [comment]
@kmrmt @vankichi could you please review VSCode settings? I'm not good at VSCode cus I'm Vimmer.
I'd appreciate if you could review.

**Comment by vankichi** [comment]
@kpango @rinx 
Unfortunately, I'm not good ad VSCode 😅 
But I tried to apply `.devcontainer/devcontainer.jon` to my local VSCode, it looks good for me.
IMO, LGTM about this setting 👍

**Review by vankichi — DISMISSED** [review]

**Comment by kpango** [comment]
seems 2 approvals I think it's okay to merge.
/rebase
/format
/approve

**Pull request merged**